File: 844.backspace-string-compare.java
import java.util.Stack;

/*
 * @lc app=leetcode id=844 lang=java
 *
 * [844] Backspace String Compare
 */

// @lc code=start
class Solution {
    public boolean backspaceCompare(String s, String t) {
        Stack<Character> word1 = new Stack<Character>();
        Stack<Character> word2 = new Stack<Character>();
        
        for (int i = 0; i < s.length(); i++) {
            char p = s.charAt(i);
            if (p == '#') {
                try {
                    
                    word1.pop();
                } catch (Exception e) {
                    continue;
                }
                continue;
            }
            word1.push(p);
        }
        for (int i = 0; i < t.length(); i++) {
            char p = t.charAt(i);
            if (p == '#') {
                try {
                    
                    word2.pop();
                } catch (Exception e) {
                    continue;
                }
                continue;
               
            }
            word2.push(p);
        }
        if (word1.size() != word2.size()) {
            return false;
        }
        while (!word1.isEmpty()) {
            if (word1.pop() != word2.pop()) {
                return false;
            }

        }
        return true;

    }
}
// @lc code=end

File: 844.backspace-string-compare.py
#
# @lc app=leetcode id=844 lang=python3
#
# [844] Backspace String Compare
#

# @lc code=start
class Solution:
    
    def find_str(self, s):
        s1 = []
        for letter in s:
            if not s1 and letter != '#' or s1 and letter != '#':
                s1.append(letter)
            elif s1:
                s1.pop()
        return s1
    
    def backspaceCompare(self, s: str, t: str) -> bool:
        s1 = self.find_str(s)
        t1 = self.find_str(t)
        print(s1,t1)
        return s1==t1
        
        
            
        
# @lc code=end

